Using proper formatting pattern for logIOException

jvanderhoek opened the pull request:
Logger.logIOException is supplying the stacktrace in the format pattern of the logger, rather than as an argument. This causes unexpected exceptions when the exception contains a '%' character (like in UnknownHostException on Mac OSX).

Fixes #613

kdavisk6 commented:
This looks good to me.  I don't see any tests included however.  Can you add one?

jvanderhoek commented:
I've added the FormatCharacterTest to LoggerTest to test the original fix.

On adding this test I discovered that the RecordingLogger Rule was not actually being applied, because the FeignException gets thrown first, skips the recording logger and then the exception gets handled by the expected exception rule. So I've added a RuleChain to enforce the expected exception handling first and then the recording logging rule. This (of course) revealed that some of the tests were actually broken, so I've fixed those.  I hope it's acceptable.

kdavisk6 commented:
@jvanderhoek This looks very good.  Great job on catching that error.  I think this is ready to be merged.
